common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidDeviceMetadata.java

```diff
@@ -81,5 +90,28 @@ public String getDeviceModel() {
     public String getManufacturer() {
         return Build.MANUFACTURER;
     }
+
+    /**
+     * Get the android device type, i.e; if it is an nGMS teams device or a mobile device
+     *
+     * @param context {@link Context}
+     * @return device type
+     */
+    public static String getAndroidDeviceTypeFromMetadata(@NonNull final Context context) {
+        final String methodTag = TAG + " :getDeviceType";
+        String defaultDeviceType = "mobileDevice";
+        try {
+            final PackageManager packageManager = context.getPackageManager();
+            final ApplicationInfo appInfo = packageManager.getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+            final Bundle metaDataBundle = appInfo.metaData;
+            final String deviceType = metaDataBundle.getString(DEVICE_TYPE, defaultDeviceType);
+            Logger.verbose(methodTag, "Setting the deviceType as " + deviceType);
+            return deviceType;
+        } catch (final PackageManager.NameNotFoundException e) {
+            // Do not throw the exception to break the auth request when getting the app's telemetry
+            Logger.warn(methodTag, "Unable to find the app's package name from PackageManager.");
```

Comment:
Should we return unknown in this case?

Reply:
If I return unknown here, I should also change the default deviceType above to unknown. I am thinking I should change it to unknown in both the places

Reply:
I mean we still need a way to handle regular modileDevice. So you basically need 3 possible outcomes from this API

- TeamsDevice
- RegularMobileDevice
- Unknown

As long as that can be achieved then we should be fine

Reply:
We can't get the value as regular mobile device. Because in non-nGMS builds.. we have a null value set for this metadata.